src/services/balancer/pools/weighted-pool.service.ts

```diff
@@ -38,6 +46,12 @@ export interface JoinPoolRequest {
 }
 
 export default class WeightedPoolService {
+  version: FactoryVersion;
+
+  constructor(public readonly _version: FactoryVersion) {
+    this.version = _version;
+  }
```

Comment:
When you define as `public readonly` it's automatically available on the instance. It's basically short hand for defining within the constructor.
```suggestion
  constructor(public readonly version: FactoryVersion) {}
```